test/json_test.go

```diff
@@ -74,6 +94,96 @@ func TestEncBuiltinJsonMarshalInt(t *testing.T) {
 	}
 }
 
+func TestEncBuiltinJsonMarshalBool(t *testing.T) {
+	s := RunServerOnPort(TEST_PORT)
+	defer s.Shutdown()
+
+	ec := NewJsonEncodedConn(t)
+	defer ec.Close()
+	ch := make(chan bool)
+
+	ec.Subscribe("json_bool", func(b bool) {
+		if !b {
+			t.Fatalf("Received test bool of '%t', wanted 'true'\n", b)
+		}
+		ch <- true
+	})
+	ec.Publish("json_bool", true)
+	if e := Wait(ch); e != nil {
+		t.Fatal("Did not receive the message")
+	}
+}
+
+func TestEncBuiltinJsonMarshalNull(t *testing.T) {
+	s := RunServerOnPort(TEST_PORT)
+	defer s.Shutdown()
+
+	ec := NewJsonEncodedConn(t)
+	defer ec.Close()
+
+	type TestType struct{}
+	ch := make(chan bool)
+
+	var testValue *TestType
+
+	ec.Subscribe("json_null", func(i interface{}) {
+		if i != nil {
+			t.Fatalf("Received test bool of '%t', wanted 'nil'\n", i)
+		}
+		ch <- true
+	})
+	ec.Publish("json_null", testValue)
+	if e := Wait(ch); e != nil {
+		t.Fatal("Did not receive the message")
+	}
+}
+
+func TestEncBuiltinJsonMarshalArray(t *testing.T) {
+	s := RunServerOnPort(TEST_PORT)
+	defer s.Shutdown()
+
+	ec := NewJsonEncodedConn(t)
+	defer ec.Close()
+
+	ch := make(chan bool)
+
+	var a = []string{"a", "b", "c"}
+
+	ec.Subscribe("json_array", func(v []string) {
+		if !reflect.DeepEqual(v, a) {
+			t.Fatalf("Received test of '%v', wanted '%v'\n", v, a)
+		}
+		ch <- true
+	})
+	ec.Publish("json_array", a)
+	if e := Wait(ch); e != nil {
+		t.Fatal("Did not receive the message")
+	}
+}
+
+func TestEncBuiltinJsonMarshalEmptyArray(t *testing.T) {
+	s := RunServerOnPort(TEST_PORT)
+	defer s.Shutdown()
+
+	ec := NewJsonEncodedConn(t)
+	defer ec.Close()
+
+	ch := make(chan bool)
+
+	var a = make([]string, 0)
```

Comment:
Would have done that too for the test above, but here seem more natural to have: `a := []string{}`, but that may just be personal preference ;-)